thavalon-webapp/src/components/Login.tsx

```diff
@@ -3,34 +3,39 @@ import ReactModal from 'react-modal';
 import { useForm } from 'react-hook-form';
 import { Link, Redirect } from 'react-router-dom';
 import { AccountManager } from '../utils/AccountManager';
-import "./Modal.scss";
+import { InputElement } from './formComponents/InputElement';
+import { FormButton } from './formComponents/FormButton';
+import "../styles/Modal.scss";
 
 interface LoginProps {
-    setLoggedIn: React.Dispatch<React.SetStateAction<boolean>>
+    setLoggedIn: React.Dispatch<React.SetStateAction<boolean>>,
+    history: any // it's any because i don't understand error "type 'history' is not a generic"
 };
 
 interface LoginData {
     "email": string,
     "password": string
 };
 
-export function Login(props: LoginProps) {
+export function Login(props: LoginProps): JSX.Element {
     // if set, register modal is open
     const [modalIsOpen, setModalIsOpen] = useState(true);
     // hook for register form
-    const {register, handleSubmit, errors} = useForm<LoginData>();
-    // state for setting if register button is disabled
-    const [disable, setDisabled] = useState(false);
+    const {register, handleSubmit} = useForm<LoginData>();
+    // state for setting if form is being submitted or not
+    const [formSubmitting, setFormSubmitting] = useState(false);
     // state for setting register error
     const [formErrorMsg, setFormErrorMsg] = useState("");
     // state for redirecting to home on successful login
     const [redirectToHome, setRedirectToHome] = useState(false);
 
     /**
-     * Called when register modal is closed.
+     * Called when register modal is closed. Redirects to most recent non-modal page.
      */
     function closeModal() {
         setModalIsOpen(false);
+        // go -1 is equal to going back to prior page, which for login will not be a modal
+        props.history.go(-1);

```

Comment:
I think we can use the `useHistory()` hook directly in the `Login` and `Register` components instead of passing it down through props